server/src/main/java/org/elasticsearch/cluster/metadata/ComposableIndexTemplate.java

```diff
@@ -104,6 +104,11 @@ public ComposableIndexTemplate(List<String> indexPatterns, @Nullable Template te
         this(indexPatterns, template, componentTemplates, priority, version, metadata, null);
     }
 
+    public ComposableIndexTemplate(Builder builder) { 
```

Comment:
If you look at other uses of the builder pattern, their `build` method calls one of the regular constructors of the class. This avoids having to add an additional constructor just for the builder.

Reply:
Agreed. Reused existing constructor.